include/extractor/extractor.hpp

```diff
@@ -100,6 +101,8 @@ class Extractor
     void WriteConditionalRestrictions(
         const std::string &path,
         std::vector<ConditionalTurnRestriction> &conditional_turn_restrictions);
+
+    size_t FindSegregatedNodes(NodeBasedGraphFactory &factory);
```

Comment:
some commentary here would be great to describe the idea of what we want to do here.